app/api/v1/endpoints/events.py

```
@@ -149,6 +162,39 @@ async def list_events(
+    subject_id: str | None = Query(None, description="Filter to this subject"),
+    _: Annotated[object, Depends(require_permission("event", "read"))] = None,
+):
+    """Server-Sent Events stream of new events for the tenant (and optionally one subject). Requires event:read."""
+    broadcaster = getattr(request.app.state, "event_stream_broadcaster", None)
+    if broadcaster is None:
+        raise HTTPException(
+            status_code=503,
+            detail="Event stream not available",
+        )
+
+    async def event_generator():
```

Ruff (0.15.5)

[warning] Missing return type annotation for private function `event_generator`

(ANN202)